From 8e1325caff2fb37ebdf2d10c5a6a2d7c27075c49 Mon Sep 17 00:00:00 2001
From: Linu Cherian <lcherian@marvell.com>
Date: Thu, 16 Aug 2018 18:23:31 +0530
Subject: [PATCH 0046/1051] soc: octeontx2: Bringup CGX LMAC links at probe
 time.

Do link up for lmac ports by default at the time of CGX
driver probe.

Signed-off-by: Linu Cherian <lcherian@marvell.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/soc/marvell/octeontx2/cgx.c | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/drivers/soc/marvell/octeontx2/cgx.c b/drivers/soc/marvell/octeontx2/cgx.c
index 869a171cdb27..ad3b1a97e398 100644
--- a/drivers/soc/marvell/octeontx2/cgx.c
+++ b/drivers/soc/marvell/octeontx2/cgx.c
@@ -599,6 +599,20 @@ static int cgx_lmac_verify_fwi_version(struct cgx *cgx)
 		return 0;
 }
 
+static void cgx_lmac_link_up(struct cgx *cgx)
+{
+	struct device *dev = &cgx->pdev->dev;
+	int i, err;
+
+	/* Do Link up for all the lmacs */
+	for (i = 0; i < cgx->lmac_count; i++) {
+		err = cgx_fwi_link_change(cgx, i, true);
+		if (err)
+			dev_info(dev, "cgx port %d:%d Link up command failed\n",
+				 cgx->cgx_id, i);
+	}
+}
+
 static int cgx_lmac_init(struct cgx *cgx)
 {
 	struct lmac *lmac;
@@ -701,6 +715,7 @@ static int cgx_probe(struct pci_dev *pdev, const struct pci_device_id *id)
 	if (err)
 		goto err_release_lmac;
 
+	cgx_lmac_link_up(cgx);
 
 	return 0;
 
-- 
2.17.1

